rakelib/update_changelog.rake

```
@@ -0,0 +1,42 @@
+  tag = args[:tag] || `git describe --tags --abbrev=0`.strip
+  anchor = "[#{tag}]"
+
+  changelog = File.read("CHANGELOG.md")
+
+  if changelog.include?(anchor)
+    puts "Tag #{tag} already documented in CHANGELOG.md"
+    next
+  end
+
+  tag_date_output = `git show -s --format=%cs #{tag} 2>&1`
+  if $?.success?
```

RuboCop (1.73)

[convention] Prefer `$CHILD_STATUS` from the stdlib 'English' module (don't forget to require it) over `$?`.

(Style/SpecialGlobalVars)